src/controllers/bulk/bulksupported.h

```diff
@@ -1,18 +1,20 @@
-// A list of supported USB bulk devices
-
 #pragma once
 
-typedef struct bulk_supported {
-    unsigned short vendor_id;
-    unsigned short product_id;
-    unsigned char in_epaddr;
-    unsigned char out_epaddr;
-    unsigned int interface_number;
-} bulk_supported_t;
+#include <cstdint>
+
+// A list of supported USB bulk devices
+
+struct bulk_supported_t {
+    std::uint16_t vendor_id;
+    std::uint16_t product_id;
+    std::uint8_t in_epaddr;
+    std::uint8_t out_epaddr;
+    std::uint32_t interface_number = 0;
+};
 
-static bulk_supported_t bulk_supported[] = {
-        {0x06f8, 0xb105, 0x82, 0x03, 0}, // Hercules MP3e2
-        {0x06f8, 0xb107, 0x83, 0x03, 0}, // Hercules Mk4
-        {0x06f8, 0xb100, 0x86, 0x06, 0}, // Hercules Mk2
-        {0x06f8, 0xb120, 0x82, 0x03, 0}, // Hercules MP3 LE / Glow
-        {0, 0, 0, 0, 0}};
+constexpr static bulk_supported_t bulk_supported[] = {
+        {0x06f8, 0xb105, 0x82, 0x03}, // Hercules MP3e2
+        {0x06f8, 0xb107, 0x83, 0x03}, // Hercules Mk4
+        {0x06f8, 0xb100, 0x86, 0x06}, // Hercules Mk2
+        {0x06f8, 0xb120, 0x82, 0x03}, // Hercules MP3 LE / Glow
+};
```

Comment:
I think the interface number should be handled like in_epaddr and out_epaddr. This is a lookup table to get these 3 device parameters for the keypair vendor_id/product_Id. It makes the table difficult to read, if the 0 is a default value.
It's just random that Hercules devices use the interface number 0 for bulk and 1 for the audio interface. 0 is no special value here, it can be also enumerated the other way around for devices of another manufacturer.

Reply:
makes sense.

Reply:
done